enumerable/socket.go

```diff
@@ -0,0 +1,62 @@
+package enumerable
+
+import "github.com/sourcenetwork/immutable"
+
+// Socket is an extention of the enumerable interface allowing the source
+// to be replaced after initial construction.
+type Socket[T any] interface {
```

Comment:
question: Can you explain why you decided to name this type `Socket`? At the moment it doesn't jibe with my current definition of a socket.

Reply:
A socket is a fixture into which other stuff may slot into.

Reply:
I find it confusing as it leads me to believe that it relates to network sockets. Don't you think?

Reply:
network sockets took the name from the same non-computing concept and I see them as being conceptually quite similar, only one is remote/cross-process, and this is not.